src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -261,16 +259,46 @@ export default class CoExpressionTab extends React.Component<
 
     private coExpressionCache: CoExpressionCache = new CoExpressionCache(
         q => ({
-            await: () => [this.props.store.entrezGeneIdToReferenceGene],
+            await: () => [
+                this.props.store.entrezGeneIdToReferenceGene,
+                this.props.store.molecularProfileIdToProfiledSamples,
+            ],
             invoke: async () => {
                 let threshold = 0.3;
                 if (q.allData) {
                     threshold = 0;
                 }
+
+                const profileXSamples = this.props.store
+                    .molecularProfileIdToProfiledSamples.result![
+                    q.profileX.molecularProfileId
+                ];
+                const profileYSamples = this.props.store
+                    .molecularProfileIdToProfiledSamples.result![
+                    q.profileY.molecularProfileId
+                ];
+
+                let dataFilter: IDataQueryFilter = {} as any;
```

Comment:
Don't change this unless there are other changes to make too, but you don't have to cast `{}` to `any`, as `IDataQueryFilter` only has optional fields.